### DISCUSSION TITLE
Global cache keys

### DISCUSSION THREAD

**Discussion by ianschmitz** [TanStack/query#2814] [Ideas]
Howdy! 

I was curious if there's any current support or plans for supporting global cache keys that would be applied to all queries for a given `QueryClient`?

The primary use case i'm thinking of is providing some global context to all queries that should invalidate cached results if it changes. For example the authenticated user's unique id or email. If the current user changes, the cache for the previous user shouldn't be used.

I could see this becoming much more important when persistors such as https://react-query.tanstack.com/plugins/createWebStoragePersistor become mainstream. Perhaps the cache invalidation/eviction would be on the developer to evict the persisted storage if the current user changes?

**Comment by TkDodo** [comment]
I think all you would need to do to get there is prefix all your queries with that information, and query key factories can help with that: https://tkdodo.eu/blog/effective-react-query-keys

if the first entry in all query keys is the user-id of the logged in user, you can act on that.
